bolt/lib/Rewrite/PseudoProbeRewriter.cpp

```diff
@@ -218,9 +220,7 @@ void PseudoProbeRewriter::updatePseudoProbes() {
         }
 
         while (CallOutputAddress != CallOutputAddresses.second) {
-          AP.second.push_back(*Probe);
-          AP.second.back().setAddress(CallOutputAddress->second);
-          Probe->getInlineTreeNode()->addProbes(&(AP.second.back()));
+          ProbeDecoder.addInjectedProbe(*Probe, CallOutputAddress->second);
```

Comment:
Why we need this change? seems it's not related to decoding pseudo probe, or is it because we allocate a fixed size of `MCDecodedPseudoProbe`, so later there is no way to add additional probe to the vector. To address this, we have to use a new container `InjectedProbeMap` to save new probes. If so, could you add comments to explain this(maybe in the definition of `InjectedProbeMap`)?

Reply:
That's required because appending to ProbeVec may cause its reallocation so that pointers to its elements will become invalid. I'll leave a comment.